cronjob/templates/_argo_cron_workflow.tpl

```diff
@@ -182,11 +182,26 @@
       {{ template "cronjob._exit_handler_healthcheck_io" . }}
       {{- end }}
       {{- end }}
-    {{- if and (.Values.ttlStrategy) (.Values.ttlStrategy.secondsAfterCompletion) }}
+
+    {{- $ttl := .Values.ttlStrategy }}
+    {{- if or $ttl.secondsAfterCompletion $ttl.secondsAfterFailure $ttl.secondsAfterSuccess }}
     ttlStrategy:
+      {{- if $ttl.secondsAfterCompletion }}
       # The second of the pod can be alive after the job is done
-      secondsAfterCompletion: {{.Values.ttlStrategy.secondsAfterCompletion}}
+      secondsAfterCompletion: {{ $ttl.secondsAfterCompletion }}
+      {{- end }}
+      {{- if $ttl.secondsAfterFailure }}
+      # The second of the pod can be alive after the job is failed
+      secondsAfterFailure: {{ $ttl.secondsAfterFailure }}
+      {{- end }}
+      {{- if $ttl.secondsAfterSuccess }}
+      # The second of the pod can be alive after the job is succeeded
+      secondsAfterSuccess: {{ $ttl.secondsAfterSuccess }}
+      {{- end }}
+    {{- else }}
+    ttlStrategy: {}
```

Comment:
should we need this? or just leave it to default?

Reply:
As per our conversation, the default behavior will result in ttlStrategy being set to nil. This is a deliberate decision intended to prevent any unexpected behavior.

Reply:
if user specifies ttlStrategy but no valid values from secondsAfterCompletion, secondsAfterFailure and secondsAfterSuccess
i think you could put an error for it
otherwise, if user doesn't specify ttlStrategy, we should not set it too , just leave it blank